OpenRA.Mods.Common/Pathfinder/HierarchicalPathFinder.cs

```diff
@@ -886,6 +899,13 @@ static Grid GetGrid(CPos cellInGrid, Grid mapBounds)
 			if (sourceAbstractCell == null)
 				return FindPath(self, new[] { source }, target, check, heuristicWeightPercentage, customCost, ignoreActor, laneBias, pathFinderOverlay);
 
+			// If the source and target belong to different domains, there is no path.
+			RebuildDomains();
+			var targetDomain = abstractDomains[targetAbstractCell.Value];
```

Comment:
Curious. Why do you use a Dictionary and not a CellLayer for `abstractDomains`? Wouldn't a cell layer perhaps be faster?

Reply:
`asbtractDomains` only contains abstract nodes as keys. It's got roughly 100x fewer entires than the number of cells on the map. The memory increase for switching to `CellLayer` would need to be justified by some testing.